komodo/yaml_file_types.py

```diff
@@ -0,0 +1,642 @@
+import argparse
+import os
+from collections import namedtuple
+from pathlib import Path
+from typing import Dict, List
+
+from ruamel.yaml import YAML
+
+komodo_error = namedtuple(
+    "KomodoError", ["package", "version", "maintainer", "depends", "err"]
+)
+report = namedtuple(
+    "LintReport", ["release_name", "maintainers", "dependencies", "versions"]
+)
+
+
+class KomodoException(Exception):
+    def __init__(self, input: komodo_error):
+        self.error = input
+
+
+MISSING_PACKAGE = "missing package"
+MISSING_VERSION = "missing version"
+MISSING_DEPENDENCY = "missing dependency"
+MISSING_MAINTAINER = "missing maintainer"
+MISSING_MAKE = "missing make information"
+MALFORMED_VERSION = "malformed version"
+MAIN_VERSION = "dangerous version (main branch)"
+MASTER_VERSION = "dangerous version (master branch)"
+FLOAT_VERSION = "dangerous version (float interpretable)"
+
+
+def _komodo_error(package=None, version=None, maintainer=None, depends=None, err=None):
+    return komodo_error(
+        package=package,
+        version=version,
+        maintainer=maintainer,
+        depends=depends,
+        err=err,
+    )
+
+
+def __reg_version_err(errs, package, version, maintainer, err=MALFORMED_VERSION):
+    return _komodo_error(
+        package=package, version=version, maintainer=maintainer, err=err
+    )
+
+
+class YamlFile(argparse.FileType):
+    def __init__(self, *args, **kwargs):
+        super().__init__("r", *args, **kwargs)
+
+    def __call__(self, value):
+        file_handle = super().__call__(value)
+        yaml = YAML()
+        yml = yaml.load(file_handle)
+        file_handle.close()
+        assert isinstance(
+            yml, dict
+        ), "File could not be parsed as yaml. Make sure the formatting is correct"
```

Comment:
For nitpicking, since the yaml parser has not thrown an exception at this point, I guess it must be valid yaml. It has to be the wrong syntax with respect to komodo files though.

Reply:
Yes, that makes sense! If it is not a yaml file, the super will already have raised an exception